File: C Language/Find_Kth_Node_From_End_LinkedList.c
//Remove Duplicates from Sorted Singly Linked List

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node * next;
    struct node * prev;
};

struct node *head = NULL;

void linkedListTraversal(struct node* ptr)
{
    ptr = head;
    while(ptr != NULL)
    {
        printf("%d ", ptr -> data);
        ptr = ptr -> next;
    }
}

void createList(int data)
{
    struct node *ptr = (struct node*)malloc(sizeof(struct node));
    ptr -> data = data;
    ptr -> next = head;	
    head = ptr;
}

struct node* findNthNode(struct node* ptr, int key)
{
    ptr = head;
    int n = 0;
    struct node* p = head;
    if(ptr == NULL)
    {
        return;
    }
    while(ptr)
    {
        ptr = ptr -> next;
        n++;
    }
    if(n >= key)
    {
        ptr = head;
        for(int i= 0; i<n-key; i++)
        {
            ptr = ptr -> next;
        }
    }
    return ptr;
}


int main()
{
   
    createList(92);
    createList(70);
    createList(70);
    createList(36);
    createList(28);
    createList(28);
    createList(12);
    printf("Linked List is\n");
    linkedListTraversal(head);
    
    struct node* result = findNthNode(head, 4);
    printf("\n\nKth node from the end is %d\n", result -> data);
    return 0;
}

File: Python/Find_Kth_Node_From_End_LinkedList.py
class Node:
	def __init__(self, data):
		self.data = data
		self.next = None
		self.prev = None

class LinkedList:
	def __init__(self):
		self.head = None
			
	def push(self, data):
		newNode = Node(data)
		newNode.next = self.head
		self.head = newNode
		
	def printList(self, ptr):
		ptr = self.head
		if ptr is None:
			print('List has no element')
			return
		else:
			temp = ptr
			while(temp):
				print(temp.data, end=' ')
				temp = temp.next
				
	def findNthNode(self, key):
		#find the nth node from end
		ptr = self.head
		n = 0
		if ptr is None:
			return
		while(ptr):
		      ptr = ptr.next
		      n += 1
		if n >= key:
			ptr = self.head
			for i in range(0, n-key):
				ptr = ptr.next
		return ptr
					
llist = LinkedList()

llist.push(92)
llist.push(70)
llist.push(36)
llist.push(28)
llist.push(12)

print("Linked List is")
llist.printList(llist.head)

result = llist.findNthNode(3)
print("\n\nKth node from the end is", result.data)